lib/tasks/result_monitor.rake

```diff
@@ -32,6 +32,51 @@ class MonitorPipelineResults
         LogUtil.log_backtrace(exception)
       end
     end
+
+    # "stalled uploads" are not pipeline jobs, but they fit in here better than
+    # anywhere else.
+    begin
+      MonitorPipelineResults.alert_stalled_uploads!
+    rescue => exception
+      LogUtil.log_err_and_airbrake("Failed to alert on stalled uploads: #{exception.message}")
+      LogUtil.log_backtrace(exception)
+    end
+
+    begin
+      MonitorPipelineResults.fail_stalled_uploads!
+    rescue => exception
+      LogUtil.log_err_and_airbrake("Failed to fail stalled uploads: #{exception.message}")
+      LogUtil.log_backtrace(exception)
+    end
+  end
+
+  def self.fail_stalled_uploads!
+    # Delay determined by doubling the time after which an alert is sent to admins.
+    Sample
+      .stalled_uploads(6.hours)
```

Comment:
I think 6 hours runs a risk of interrupting uploads that are still going. How about 18 hours for people with slow networks that leave the browser open overnight?

Reply:
👍